### PR TITLE
Localized numbers, and dates

### LINKED ISSUE #831: localize number, for example hijri prayer time
image

using the https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleTimeString method

### PR DESCRIPTION
**Pull request by mahmoudfelfel**
### Summary
Added localization for numbers and dates.
Note: Plurals are still not very accurate in some places due to the stitching we are doing between strings and numbers but we can fix that in another PR.

Closes #831 

### Screenshots
Screen Shot 2021-11-29 at 4 00 46 AM
Screen Shot 2021-11-29 at 1 16 58 AM
Screen Shot 2021-11-29 at 1 17 28 AM
Screen Shot 2021-11-29 at 1 36 36 AM
Screen Shot 2021-11-29 at 2 15 22 AM
Screen Shot 2021-11-29 at 2 19 36 AM
Screen Shot 2021-11-29 at 2 23 42 AM
Screen Shot 2021-11-28 at 11 30 25 PM
Screen Shot 2021-11-28 at 11 31 57 PM
Screen Shot 2021-11-28 at 11 36 40 PM
Screen Shot 2021-11-28 at 11 36 55 PM
Screen Shot 2021-11-28 at 11 37 10 PM

### CODE REVIEW

**Comment by muhajirdev** [comment]
image

image

File: src/components/HomePage/PrayerTimes/PrayerTimes.tsx [hunk 1]
```diff
@@ -158,19 +156,9 @@ const PrayerTimes = () => {
...
-  // Different language have different format to show the date, so we need to "format" it.
-  // For example in Indonesia we say "12 Muharram 1443" instead of "Muharram 12, 1443"
-  return t('hijri-date.format', {
-    date: hijriDate.date,
-    month,
-    year: hijriDate.year,
-  });
+const useHijriDateFormatter = () => {
+  const { lang } = useTranslation('home');
+  return toLocalizedDate(new Date(), lang, { calendar: 'islamic', dateStyle: 'full' });
```
Comment (muhajirdev) [inline src/components/HomePage/PrayerTimes/PrayerTimes.tsx:161]: We previously tried with local data too with 

```javascript
new Intl.DateTimeFormat('en-TN-u-ca-islamic', {day: 'numeric', month: 'long',weekday: 'long',year : 'numeric'}).format(Date.now());
```

If I remember correctly, i didn't show the right data. But it seems to be correct now. hmm. maybe need to double check this one

File: src/components/Fonts/FontPreLoader.tsx [hunk 2]
```diff
@@ -10,7 +10,7 @@ const LOCALE_PRELOADED_FONTS = {
     { type: 'font/woff2', location: 'ProximaVara/proxima_vara_semibold.woff2' },
     { type: 'font/woff2', location: 'ProximaVara/proxima_vara_bold.woff2' },
   ],
-  ar: [{ type: 'font/woff2', location: 'AlJazeera/Al-Jazeera-Regular.woff2' }],
```
Comment (muhajirdev) [inline src/components/Fonts/FontPreLoader.tsx:13]: I am assuming Noto has a better numeric support?

Comment (osamasayed) [reply src/components/Fonts/FontPreLoader.tsx:13]: Yes correct

**Pull request merged**